apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/sidebar-modal-opener.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { Button, Modal } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { PageTemplatesPlugin } from '../index';
+import TemplateSelectorItem from './template-selector-item';
+import replacePlaceholders from '../utils/replace-placeholders';
+/* eslint-enable import/no-extraneous-dependencies */
+
+class SidebarModalOpener extends Component {
+	state = {
+		isTemplateModalOpen: false,
+		isWarningOpen: false,
+	};
+
+	toggleTemplateModal = () => {
+		this.setState( {
+			isTemplateModalOpen: ! this.state.isTemplateModalOpen,
+			isWarningOpen: false,
+		} );
+	};
+
+	toggleWarningModal = () => {
+		this.setState( { isWarningOpen: ! this.state.isWarningOpen } );
+	};
+
+	getLastTemplateUsed = () => {
+		if ( ! this.props.templateUsedSlug || this.props.templateUsedSlug === 'blank' ) {
+			return this.props.templates[ 1 ];
+		}
+		return this.props.templates.filter( temp => temp.slug === this.props.templateUsedSlug )[ 0 ];
+	};
+
+	// The TemplateSelectorItem wants a function to run when clicked, we want it to do nothing from here.
+	onSelectOverride = () => {};
+
+	render() {
+		const { slug, title, preview, previewAlt } = this.getLastTemplateUsed();
+		const { templates, vertical, segment, siteInformation } = this.props;
+
+		return (
+			<div className="sidebar-modal-opener">
+				<TemplateSelectorItem
+					id="sidebar-modal-opener__last-template-used-preview"
+					value={ slug }
+					label={ replacePlaceholders( title, siteInformation ) }
+					staticPreviewImg={ preview }
+					staticPreviewImgAlt={ previewAlt }
+					onSelect={ this.onSelectOverride }
+				/>
+
+				<Button
+					isPrimary
+					onClick={ this.toggleWarningModal }
+					className="sidebar-modal-opener__button"
+				>
+					{ __( 'Change Layout' ) }
+				</Button>
+
+				{ this.state.isTemplateModalOpen ? (
+					<PageTemplatesPlugin
+						shouldPrefetchAssets={ false }
+						templates={ templates }
+						vertical={ vertical }
+						segment={ segment }
+						toggleTemplateModal={ this.toggleTemplateModal }
+						isPromptedFromSidebar
+					/>
+				) : null }
+
+				{ this.state.isWarningOpen ? (
```

Comment:
cc @shaunandrews. What if we showed the modal _after_ choosing a template? Clicking "change template" immediately opens the selector, then we have a cancel button in the template selector. And clicking "use template" shows this modal instead.

Reply:
While this makes more sense to me, I don't think it's that much of an improvement to justify blocking this PR on it, especially if implementing that might be more tricky in terms of modifying the existing SPT content insertion flow and functionality. We can address it in a follow-up if it turns out to be important in usability testing.